woocommerce/woocommerce-ios
# Pass anonymous ID to ExPlat configuration for authenticated users

Closes: #7435 


- [x] ⚠️ Please also review https://github.com/Automattic/Automattic-Tracks-iOS/pull/213 ⚠️ 
- [x] ⚠️ Update the pod version after the PR above is merged and deployed ⚠️ 

### Description


Previously, we passed `nil` to the anonymous ID parameter in the ExPlat configuration and we observed a high crossover rate where the same logged-in user has been assigned to more than one variation for the same A/A test. From @aaronyan's investigation pbxNRc-1S0-p2#comment-3688, the `nil` anonymous ID might have caused a high crossover rate because the anonymous ID is prioritized over the authenticated user ID when assigning a variation for an A/B experiment.

> Once assignment requests include the anon_id, then I think we should see a lot less crossovers! The anon_id will be given priority on the assignment side, so the user will be assigned the same variation that was originally given to them while they were anonymous.

This PR passed the persisted anonymous ID to the new parameter `anonymousID` in all the `TracksService.switchToAuthenticatedUser` calls.

### Testing instructions


Prerequisites:
In Xcode, set a breakpoint at `ExPlatService` L47 where `anon_id` is set in the URL request

- Launch the app, log out if needed
- Relaunch the app --> make a note of the anon ID in the Xcode console
- Log in to the store picker screen --> the anon ID in the Xcode console should be set to the same as before
- Relaunch the app (logged-in state) --> the anon ID in the Xcode console should be set to the same as before

---
